### PR TITLE
Fixes #1012 - Add support for lambda layers

### LINKED ISSUE #1012: Support for Lambda layers
It should be possible to add Lambda layers to a Lambda without having to provide your own cloudformation.template.yml file.   

/cumulus/packages/deployment/app/cloudformation.template.yml:642
```
{{#each lambdas}}
  {{@key}}LambdaFunction:
    Type: AWS::Lambda::Function
    Properties:
      Code:
        S3Bucket: {{this.bucket}}
        S3Key: {{this.remote}}
      FunctionName: {{../stackName}}-{{@key}}
    {{#if this.layers}}
      Layers:
      {{#each this.layers}}
        - {{this}}
      {{/each}}
    {{/if}}
      Environment:
        Variables:
```

The addition is from `{{#if this.layers}}` to it's ending `{{/if}}`

Then layers can be specified in `lambdas.yml` such as:
```
MyLambda:
  runtime: python3.6
  handler: lambda_function.lambda_handler
  timeout: 300
  memory: 128
  s3Source:
    bucket: '{{buckets.internal.name}}'
    key: deploy/mylambda/0.1.0.zip
  layers:
    - arn:aws:lambda:us-west-2:552188055668:layer:geolambda:1
```

I could make a PR for this, but I'm not sure that the contributing guidelines are still valid given that issues aren't tracked within GitHub.

### PR DESCRIPTION
**Pull request by garysieling**
**Summary:** Summary of changes

Addresses [1012]: Support for lambda layers

## Changes

* Conditionally listens for a lambda layers parameter. NOTE: I only tested with one layer as that's what we needed. It's a little tricky to get arrays to work in the kes substitution variables so it's possble that more than one won't work.

## PR Checklist

- [x] Update CHANGELOG
- [ ] Unit tests
- [x] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: packages/deployment/app/cloudformation.template.yml [hunk 1]
```diff
@@ -412,6 +412,10 @@ Resources:
         S3Bucket: {{this.bucket}}
         S3Key: {{this.remote}}
       FunctionName: {{../prefix}}-{{@key}}
+    {{# if this.layers}}
```
Comment (markdboyd) [inline packages/deployment/app/cloudformation.template.yml]: Can we try updating this to handle one to many layers? Probably like so:

```
{{# if this.layers}}
  Layers:
  {{#each this.layers}}
    - {{this.layers}}
  {{/each}}
{{/if}}
```

Then your `lambdas.yml` entry would look like something like:

```
Lambda1:
  layers:
    - layer1
    - layer2
```

Comment (laurenfrederick) [reply packages/deployment/app/cloudformation.template.yml]: @markdboyd I agree with your comment

I think if `layers` looks like this:
```
layers: 
  - layer1arn
  - layer2arn
```

you'd want this in the template:
```
{{# if this.layers}}
  Layers:
  {{#each this.layers}}
    - {{this}}
  {{/each}}
{{/if}}
```
Swapping out `this.layers` for `this`

@garysieling Where would the actual lambda layers be defined? Do you have an example?

**Comment by garysieling** [comment]
@laurenfrederick 

An example layer is this one, which has a bunch of libraries pre-installed -
https://github.com/developmentseed/geolambda

`layers:
    - arn:aws:lambda:us-west-2:552188055668:layer:geolambda:1`

**Comment by garysieling** [comment]
@markdboyd @laurenfrederick I tested with this syntax and it appears to work fine (i.e. it deploys fine - I don't have a use case to test for multiple layers) - updated the PR

**Pull request merged**